src/include/ZividPython/Wrappers.h

```diff
@@ -13,6 +16,49 @@
 
 namespace ZividPython
 {
+    namespace
+    {
+        std::string toSnakeCase(const std::string capitalized)
+        {
+            if(capitalized.empty())
+            {
+                throw std::invalid_argument{ "String is empty." };
+            }
+
+            if(!isupper(capitalized[0]))
+            {
+                throw std::runtime_error{ "First character of string: '" + capitalized + "' is not capitalized" };
+            }
+            std::stringstream ss;
+            ss << char(tolower(capitalized[0]));
+
+            for(std::string::size_type i = 1; i < capitalized.size(); ++i)
+            {
+                if(isupper(capitalized[i]))
+                {
+                    std::string::size_type previous = i - 1;
+                    std::string::size_type next = i + 1;
+
+                    if(isupper(capitalized[previous]) && (next < capitalized.size() && isupper(capitalized[next]))
+                       || (next >= capitalized.size()))
+                    {
+                        ss << char(tolower(
+                            capitalized[i])); // if surrounded by capital case (looking at the B character): ABC -> abc
```

Comment:
It is nice to have the comment before the code. Else you read the code, figure out what it does, then when you reach the comment it is redundant.

Also trailing comment makes formatting weird.

I would put the comment all the way before the if, because that is actually what you are describing.